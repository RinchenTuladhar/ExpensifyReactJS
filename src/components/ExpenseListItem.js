import React from "react";
import { Link } from "react-router-dom";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={"/edit/" + id}>Edit</Link>
    <p>{description}</p>
    <p>
      {amount} - {createdAt}
    </p>
  </div>
);

export default ExpenseListItem;
