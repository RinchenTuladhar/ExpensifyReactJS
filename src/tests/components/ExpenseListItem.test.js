import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";
const item = {
  id: 1,
  description: "Duffle bag",
  amount: "10",
  createdAt: 0,
};

test("should render ExpenseListItem", () => {
  const wrapper = shallow(<ExpenseListItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
