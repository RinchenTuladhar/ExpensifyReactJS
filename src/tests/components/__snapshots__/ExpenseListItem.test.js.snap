// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    Edit
  </Link>
  <p>
    Bum bag
  </p>
  <p>
    $25.00
    -
    1st January 1970
  </p>
</div>
`;
